scripts/mass_user_emails.py

```diff
@@ -0,0 +1,235 @@
+import os
+import time
+
+import boto3
+from dateutil.relativedelta import relativedelta
+from django.contrib.auth.models import User
+from django.db.models.functions import Lower
+from django.utils import timezone
+
+MAX_SEND_ATTEMPTS = 3
+RETRY_WAIT_TIME = 30  # in seconds
+FROM_ADDRESS = ''
+EMAIL_SUBJECT = ''
+EMAIL_TEMPLATE_NAME = ''
+
+# Path to a valid .html file to use for the email
+EMAIL_HTML_FILENAME = ''
+
+# Path to a valid .txt file to use for the email text content
+EMAIL_TEXT_FILENAME = ''
+
+
+# We don't want to use all of our available email sends; some need to be reserved for other uses (password resets, etc.)
+# So we send emails until we've sent ( 24 hour sending capacity - RESERVE_EMAIL_COUNT ) emails
+RESERVE_EMAIL_COUNT = 4000
+
+# Whether this is a marketing email or a transactional email.
+# Marketing emails aren't sent to addresses that have registered spam complaints
+# Transactional emails ignore spam complaints
+IS_MARKETING_EMAIL = True
+
+# Maximum number of emails to send in one use of the script
+# Set to 0 to send as many emails as SES will allow
+MAX_SEND_LIMIT = 0
+
+# Only sends to users who have logged in since this date
+# Needs to be a Date object
+ACTIVE_SINCE = timezone.now() - relativedelta(years=1)
+
+start_time = time.time()
+
+
+def run(*args):
+    # To run the script in test mode, use './manage.py runscript mass_user_emails --script-args test'
+    test_mode = 'test' in args
+    # Use the 'force' arg to send emails even to users that have received the email before
+    force_send = 'force' in args
+
+    aws_region_name = os.environ.get('AWS_SES_REGION_NAME') or os.environ.get(
+        'AWS_S3_REGION_NAME'
+    )
+    ses = boto3.client('ses', region_name=aws_region_name)
+
+    remaining_sends = MAX_SEND_LIMIT
+    quota = ses.get_send_quota()
+    # if Max24HourSend == -1, we have unlimited daily sending quota
+    if quota['Max24HourSend'] >= 0:
+        print(
+            f"{int(quota['SentLast24Hours'])} of {int(quota['Max24HourSend'])} *total* emails sent in the last 24 hours"
+        )
+        quota_sends = int(
+            quota['Max24HourSend']
+            - quota['SentLast24Hours']
+            - RESERVE_EMAIL_COUNT
+        )
+        if quota_sends <= 0:
+            quit(f'already over sending limit; exiting...')
+        remaining_sends = (
+            remaining_sends >= 0
+            and min(quota_sends, remaining_sends)
+            or quota_sends
+        )
+        string_sends = remaining_sends >= 0 and remaining_sends or "unlimited"
```

Comment:
I have to admit, this took me an embarrassing amount of time to wrap my head around. I don't know if it's worth changing such well-tested code, but I'd offer for future consideration something like
```python
string_sends = remaining_sends if remaining_sends > 0 else "unlimited"
```